File: python3/binary_tree_inorder_traversal.py
from .model import TreeNode

"""
Space   : O(1)
Time    : O(n)
"""

class Solution:
    def getInorder(self, root: TreeNode) -> List[int]:
        res = []
        
        if root:
            res += self.inorderTraversal(root.left)
            res.append(root.val)
            res += self.inorderTraversal(root.right)
            
        return res

File: golang/binary_tree_inorder_traversal.go
package main

import "golang/model"

func inorderTraversal(root *model.TreeNode) []int {
	var res []int

	if root != nil {
		res = append(res, inorderTraversal(root.Left)...)
		res = append(res, root.Val)
		res = append(res, inorderTraversal(root.Right)...)
	}

	return res
}
